pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go

```diff
@@ -55,19 +55,28 @@ func ExtractPod(p *corev1.Pod) *model.Pod {
 	podModel.Status = computeStatus(p)
 	podModel.ConditionMessage = getConditionMessage(p)
 
-	for _, c := range p.Spec.Containers {
+	podModel.ResourceRequirements = GetModelResourceRequirements(p.Spec.Containers, p.Spec.InitContainers)
+
+	return &podModel
+}
+
+func GetModelResourceRequirements(containers []corev1.Container, initContainers []corev1.Container) []*model.ResourceRequirements {
```

Comment:
Suggestion on the method name: to follow the rest of the package I think it would make sense to rename it `ExtractResourceRequirements` (the model part is redundant with the fact we expect to return models from k8s objects in that package and we have named most of our extraction methods `extract<Foo>` or `convert<Foo>` AFAICT).

Also do you mind adding a small comment since that function is exposed ?

Reply:
That makes sense! 
Let me rename it, the problem with passing in `corev1.PodTemplateSpec` is that `pods` are using `PodSpec` while the other workloads are using `PodTemplateSpec`.

So the choices I see are:
- keep using 2 different containerTypes as input
- use 2 different methods
- use an interface and convert when needed

I prefer to keep the first choice, let me know if you think otherwise!

Reply:
the two methods approach is great thanks for updating 👍